### PR TITLE
Fix for 1172

### PR DESCRIPTION
**Pull request by hvanbakel**
PR for #1172

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs [hunk 1]
```diff
@@ -75,8 +75,19 @@ private static SyntaxNode ReformatBlockAndParent(Document document, SyntaxNode s
                 newParentLastToken = newParentLastToken.WithTrailingTrivia(newTrailingTrivia);
             }
 
+            var parentNextToken = block.CloseBraceToken.GetNextToken();
+
+            var nextTokenLine = parentNextToken.GetLineSpan().StartLinePosition.Line;
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs]: :bulb: This is available via an extension method in `LocationHelpers`:

```csharp
parentNextToken.GetLine();
```

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs [hunk 2]
```diff
@@ -75,8 +75,19 @@ private static SyntaxNode ReformatBlockAndParent(Document document, SyntaxNode s
...
+            var blockCloseLine = block.CloseBraceToken.GetLineSpan().EndLinePosition.Line;
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs]: :bulb:

```csharp
block.CloseBraceToken.GetEndLine();
```

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs [hunk 3]
```diff
@@ -75,8 +75,19 @@ private static SyntaxNode ReformatBlockAndParent(Document document, SyntaxNode s
                 newParentLastToken = newParentLastToken.WithTrailingTrivia(newTrailingTrivia);
             }
 
+            var parentNextToken = block.CloseBraceToken.GetNextToken();
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs]: :question: What happens if `CloseBraceToken` is the last token in the file? For example, what happens if the following code is the entire file?

```csharp
class TypeName
{
  void MethodName()
  {
    if (false) { return; }
```

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs [hunk 4]
```diff
@@ -418,6 +418,49 @@ public void Bar()
...
+        do
+        {
+            Bar();
+        }
+        while (false);
+    }
+}
+";
+
+            DiagnosticResult[] expectedDiagnostics =
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs]: :bulb: You only need an array for tests producing multiple diagnostics.

```csharp
DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(7, 12);
```

**Comment by sharwell** [comment]
I left a question and a few comments in the diff.

In addition, can you amend this to use your name instead of "connected-web-development" as the author of the commit(s)? Thanks!

**Comment by sharwell** [comment]
@hvanbakel I marked this **do not merge** just so it's easy for me to see that I finished reviewing it and it's waiting for feedback. Once you have a look at the feedback let me know and I'll remove that label. :smile:

**Comment by hvanbakel** [comment]
Looking at it right now

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs (hunk 3 above)
Comment (hvanbakel) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1501CodeFixProvider.cs]: Added a test for it, works fine

**Comment by sharwell** [comment]
Rebased and merged manually in 1498eaf672f2cf01d1620d0e2c6538f53f22b81f.

Thank you so much! :tada:

**Pull request closed without merging**